### ISSUE TITLE
mypy unhappy with `from io import StringIO`

### ISSUE THREAD

**Issue by acdimalev** [evhub/coconut#385]
The simple statement `from io import StringIO` results in an error when running mypy checks.

```
$ coconut error.coco -s
Compiling         error.coco ...
Compiled to       error.py .
$ coconut error.coco -s --mypy
Compiling         error.coco ...
Compiled to       error.py .
error.py:22: error: Incompatible import of "StringIO" (imported name has type "Type[StringIO]", local name has type "T
ype[StringIO[Any]]")
$ cat error.coco
from io import StringIO
```

I did some investigation on this front.  It looks like mypy is not terribly fond of imports in conditional blocks.

```
$ cat mypy-test-1.py
if False:
    from StringIO import StringIO
else:
    from io import StringIO
$ mypy --py2 mypy-test-1.py
mypy-test-1.py:4: error: Incompatible import of "StringIO" (imported name has type "Type[StringIO]", local name has type "Type[StringIO[Any]]")
$ 
$ cat mypy-test-2.py
from io import StringIO
$ mypy --py2 mypy-test-2.py
$ 
$ cat mypy-test-3.py
from StringIO import StringIO
from io import StringIO
$ mypy --py2 mypy-test-3.py
mypy-test-3.py:2: error: Incompatible import of "StringIO" (imported name has type "Type[StringIO]", local name has type "Type[StringIO[Any]]")
```

**Comment by evhub** [comment]
@acdimalev I believe this is currently fixed on `coconut-develop`--if you `pip install -U coconut-develop`, does that fix the problem?

**Comment by acdimalev** [comment]
Gave it a try.  No such luck.

```
$ make
coconut error.coco -s --mypy
Compiling         error.coco ...
CoconutWarning: found unused import: 'StringIO' (disable --strict to dismiss)
Compiled to       error.py .
error.py:22: error: Incompatible import of "StringIO" (imported name has type "Type[StringIO]", local name has type "Type[StringIO[Any]]")
Exiting due to MyPy error.
make: *** [Makefile:7: error.py] Error 1
$ cat error.coco
from io import StringIO
$ coconut --version
Coconut: Version 1.3.1-post_dev18 [Dead Parrot] running on Python 3.6.4
```

I assume the `type: ignore` can be removed from the `specific.coco` test once this is working without problem.

https://github.com/evhub/coconut/blob/0851ee6505a2152c4841d37b5642a9b80709d2ff/tests/src/cocotest/agnostic/specific.coco#L1

**Comment by evhub** [comment]
@acdimalev As a simple temporary fix, you should be able to pass an explicit `--target` to make this go away.

**Comment by evhub** [comment]
@Hibou57 Under `--target universal` (the default target), Coconut will compile `from io import StringIO` into
```python
if sys.version_info < (2, 7):
    from StringIO import StringIO
else:
    from io import StringIO
```
which can cause mypy issues, as per the original issue here. That being said, just doing `import io` as in the snippet you posted should pass through without any changes, so if that's causing mypy issues the culprit must be something else. To check, I put the snippet you posted in an otherwise empty Python file and ran mypy on it, and I got the same error, which implies that it certainly can't be a Coconut issue, so I would definitely recommend raising a `typeshed` issue with your snippet.

**Comment by Hibou57** [comment]
Oops, I juste noticed I did not post on Mypy. Apologies …

**Comment by evhub** [comment]
Had to do some craziness to get around https://github.com/python/mypy/issues/6189, but this should now be fixed on `coconut_develop>=2.0.0-post_dev11`.
